src/main/java/com/tribesbackend/tribes/controllers/usercontrollers/UserRestController.java

```diff
@@ -0,0 +1,113 @@
+    UserTRepository userTRepository;
+    UserModelHelpersMethods userMethods;
+    UserCrudService userCrudService;
+    KingdomRepository kingdomRepo;
+
+
+    @Autowired
+    public UserRestController(UserTRepository userTRepository, UserModelHelpersMethods userMethods,
+                              UserCrudService userCrudService, KingdomRepository kingdomRepo) {
+        this.userTRepository = userTRepository;
+        this.userMethods = userMethods;
+        this.userCrudService = userCrudService;
+        this.kingdomRepo = kingdomRepo;
+    }
+
+    @PostMapping(value = "/register")
+    public ResponseEntity registerUser(@Validated @RequestBody RegistrationInputJson regjson) {
+        TribesUser newUser = new TribesUser(regjson.getUsername(), regjson.getPassword());
+
+        if (userMethods.usernameAlreadyTaken(newUser)) {
+            return new ResponseEntity(ErrorMessagesMethods.usernameAlreadyTaken(), HttpStatus.CONFLICT);
+        } else
+            userTRepository.save(newUser);
+        Kingdom newKingdom = new Kingdom(regjson.getKingdom(), newUser);
+        newUser.setKingdom(newKingdom);
+        System.out.println(newKingdom.getName());
+        System.out.println(newKingdom.getId());
+        kingdomRepo.save(newKingdom);
+        System.out.println(newUser.getId());
+        System.out.println(newUser.getUsername());
+        System.out.println(newKingdom.getId());
+        return new ResponseEntity(new RegistrationResponseJson(newUser.getId(), newUser.getUsername(),
+                newKingdom.getId(), "No avatar yet", 0), HttpStatus.OK);
+
+    }
+
+    @PostMapping(value = "/login")
+    public ResponseEntity loginUser(@RequestBody TribesUser tribesUser) {
+        if (tribesUser.getUsername() == null || tribesUser.getUsername().isEmpty() ||
+                tribesUser.getPassword() == null || tribesUser.getPassword().isEmpty()) {
+            return new ResponseEntity(ErrorMessagesMethods.jsonFieldIsEmpty(tribesUser), HttpStatus.BAD_REQUEST);
+
+
+        } else if (!userTRepository.findTribesUserByUsername(tribesUser.getUsername()).isPresent()) {
+            return new ResponseEntity(ErrorMessagesMethods.notSuchUser(tribesUser.getUsername()), HttpStatus.UNAUTHORIZED);
+        } else if (userTRepository.findTribesUserByUsername(tribesUser.getUsername()).get().getPassword().equals(tribesUser.getPassword())) {
+            userTRepository.findTribesUserByUsername(tribesUser.getUsername()).get().setLoggedIn(true);
+            userTRepository.save(userTRepository.findTribesUserByUsername(tribesUser.getUsername()).get());
+            return new ResponseEntity(
+                    new OKstatus(
+                            JWT.create()
+                                    .withSubject(tribesUser.getUsername())
+                                    .withExpiresAt(new Date(System.currentTimeMillis() + EXPIRATION_TIME))
+                                    .sign(HMAC512(SecurityConstants.SECRET.getBytes())))
```

Comment:
@Kopsova extract this method into a `JWTService` which has two methods:
- createToken(username)
- extractUsername(token)

Reply:
the second should be used in the filter

Reply:
done